fsgs/FSGSDirectories.py

```diff
@@ -31,6 +70,56 @@ def get_base_dir(cls):
         return fsboot.base_dir()
 
     @classmethod
+    def _get_real_dir(cls, dir_name):
+        if dir_name not in cls.DIRNAME_TO_REALPATH:
+            raise NotImplementedError(
+                "'{}' not managed in {}".format(dir_name,
+                                                cls.DIRNAME_TO_REALPATH))
+        return (cls.portable_dir(dir_name) or
+                os.path.join(cls.get_base_dir(),
+                             cls.DIRNAME_TO_REALPATH[dir_name]))
+
+    @classmethod
+    def ensure_dir_exists(cls, dir_name):
```

Comment:
this utility method basically checks whether the directory exists and if not it clears the cache of the relevant function, so when it is invoked it actually runs.